### PR TITLE
chore: release 0.44.0

### PR DESCRIPTION
**Pull request by SdgJlbl**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -42,10 +44,10 @@ scores, intermediate_state_train, intermediate_state_agg = simulate_experiment(
 - BREAKING: rename `test_data_sample_keys`, `test_tasks` and `register_test_operations`, `tasks` to `data_sample_keys` and `register_operations` in `TestDataNodes` ([#185](https://github.com/Substra/substrafl/pull/185))
 - BREAKING: `InputIdentifiers` and `OutputIdentifiers` move from `substrafl.nodes.node` to `substrafl.nodes.schemas` ([#185](https://github.com/Substra/substrafl/pull/185))
 - Switch to python-slim as base image, instead of substra-tools ([#197](https://github.com/Substra/substrafl/pull/197))
+- Dropped support for Python 3.8
```
Comment (guilhem-barthes) [inline CHANGELOG.md:47]: Maybe add a link to this PR?

File: CHANGELOG.md [hunk 2]
```diff
@@ -42,10 +44,10 @@ scores, intermediate_state_train, intermediate_state_agg = simulate_experiment(
...
 
 ### Fixed
 - Numerical stability of the `NewtonRaphson` strategy is improved by symmetrizing the Hessian
-([#196](https://github.com/Substra/substrafl/pull/196))
```
Comment (guilhem-barthes) [inline CHANGELOG.md:48]: Why did you remove this link?

File: CHANGELOG.md (hunk 1 above)
Comment (SdgJlbl) [reply CHANGELOG.md:47]: It was done during the release 🙈

**Review by guilhem-barthes — APPROVED** [review]
